pkg/gofr/http/middleware/logger.go

```diff
@@ -107,6 +112,46 @@ func Logging(logger logger) func(inner http.Handler) http.Handler {
 	}
 }
 
+func getCorrelationAndSpanID(request *http.Request) (r *http.Request, correlationID, spanID string) {
+	correlationID = request.Header.Get("X-Correlation-ID")
+
+	var ctx context.Context
+
+	if correlationID == "" {
+		ctx, correlationID, spanID = getIDs(request.Context())
+		// returning deep copy of request with added context
+		request = request.Clone(ctx)
```

Comment:
can we put the context in the request? Clone would be a complex operation - might not give the desired perforamnce results